props/saber_BC_buttons.h

```diff
@@ -477,738 +1495,1272 @@ class SaberBCButtons : public PROP_INHERIT_PREFIX PropBase {
+#endif
+    // Allow beep to be heard.
     if (scroll_presets_timer_.timerCheck() && scroll_presets_) {
         SaberBase::DoEffect(EFFECT_NEWFONT, 0);
     }
+    if (twist_timer_.timerCheck() && saved_twist_) {
+      DoSavedTwist();
+    }
+    quote_delay_timer_.timerCheck();
   }  // Loop()
 
 #ifdef SPEAK_BLADE_ID
   void SpeakBladeID(float id) override {
     if (&SFX_mnum) {
       sound_library_.SayNumber(id, SAY_WHOLE);
     } else {
-      PVLOG_NORMAL << "No mnum.wav number prompts found.\n";
+      PVLOG_NORMAL << "** No mnum.wav number prompts found.\n";
       beeper.Beep(0.25, 2000.0);
     }
   }
 #endif
 
-// Revert colorchange witout saving (reset to Variation == 0)
-  void ResetColorChangeMode() {
-    if (!current_style()) return;
-      PVLOG_NORMAL << "Reset Color Variation\n";
-      SetVariation(0);
-      PVLOG_NORMAL << "Color change mode done, variation = " << SaberBase::GetCurrentVariation() << "\n";
-      SaberBase::SetColorChangeMode(SaberBase::COLOR_CHANGE_MODE_NONE);
-  }
-
-// Volume Menu
-    void VolumeMenu() {
-    if (!mode_volume_) {
-      current_menu_angle_ = fusor.angle2();
-      mode_volume_ = true;
-      if (SFX_vmbegin) {
-        sound_library_.SayEnterVolumeMenu();
-      } else {
-        beeper.Beep(0.1, 1000);
-        beeper.Beep(0.1, 2000);
-        beeper.Beep(0.1, 3000);
-      }
-      PVLOG_NORMAL << "Enter Volume Menu\n";
-      SaberBase::DoEffect(EFFECT_VOLUME_LEVEL, 0);
-    } else {
-      mode_volume_ = false;
-      if (SFX_vmend) {
-        sound_library_.SayVolumeMenuEnd();
-      } else {
-        beeper.Beep(0.1, 3000);
-        beeper.Beep(0.1, 2000);
-        beeper.Beep(0.1, 1000);
-      }
-      PVLOG_NORMAL << "Exit Volume Menu\n";
+  void DetectMenuTurn() {
+    float a = fusor.angle2() - current_menu_angle_;
+    if (isPointingUp()) return;
+    // Keep the rotational angle within range of -180 to 180.
+    if (a > M_PI) a-= M_PI * 2;
+    if (a < -M_PI) a+= M_PI * 2;
+
+    if (a < -M_PI / 6) {
+      CheckSavedTwist(EVENT_TWIST_LEFT);
+    } else if (a > M_PI / 6) {
+      CheckSavedTwist(EVENT_TWIST_RIGHT);
     }
   }
 
-  void VolumeUp() {
-    SaberBase::DoEffect(EFFECT_VOLUME_LEVEL, 0);
-    current_menu_angle_ = fusor.angle2();
-    if (dynamic_mixer.get_volume() < VOLUME) {
-      dynamic_mixer.set_volume(std::min<int>(VOLUME + VOLUME * 0.1,
-        dynamic_mixer.get_volume() + VOLUME * 0.10));
-      if (SFX_volup) {
-        hybrid_font.PlayPolyphonic(&SFX_volup);
-      } else {
-        beeper.Beep(0.10, 2000);
-        beeper.Beep(0.20, 2500);
-      }
-      PVLOG_NORMAL << "Volume Up - Current Volume: ";
-      PVLOG_NORMAL << dynamic_mixer.get_volume() << "\n";
-    } else {
-      QuickMaxVolume();
+  void CheckSavedTwist(uint32_t event) {
+    if (!saved_twist_) {
+      // Save the current twist and start the timer if no twist is saved
+      saved_twist_ = event;
+      twist_timer_.trigger(DOUBLE_TWIST_TIMEOUT);
```

Comment:
What is a "double twist" ?

Reply:
I know, that's a bad choice. But "TWIST_TIMEOUT" seems unclear though, even though it's not specified LEFT or RIGHT.

It's just "regular twist", the 2-direction gesture.
Is "regular twist" better? 
"bi-directional twist"?
"normal twist"
I'll go with REGULAR_TWIST_TIMEOUT. Let me know otherwise.